src/System.Private.Uri/src/System/DomainNameHelper.cs

```diff
@@ -203,30 +204,31 @@ internal static unsafe bool IsValidByIri(char* name, ushort pos, ref int returne
             return true;
         }
 
-        internal static string IdnEquivalent(string hostname)
+        [return: NotNullIfNotNull("hostname")]
+        internal static string? IdnEquivalent(string hostname)
```

Comment:
Given this attribute and the fact that hostname is non-nullable should the result be always non-nullable or hostname nullable?

Reply:
Yup. Fixed.